src/webview/createOrShowKedroVizPanel.ts

```diff
@@ -0,0 +1,55 @@
+import * as vscode from 'vscode';
+import KedroVizPanel from './vizWebView';
+import { sendHeapEventWithMetadata } from '../common/telemetry';
+import { LanguageClient } from 'vscode-languageclient/node';
+import { checkKedroViz, installKedroViz, updateKedroVizPanel } from '../common/utilities';
+
+async function createOrShowKedroVizPanel(
+    context: vscode.ExtensionContext,
+    lsClient: LanguageClient | undefined,
+): Promise<void> {
+    KedroVizPanel.createOrShow(context.extensionUri);
+    updateKedroVizPanel(lsClient);
+    await sendHeapEventWithMetadata('kedro.runKedroViz', context);
```

Comment:
```suggestion
   sendHeapEventWithMetadata('kedro.runKedroViz', context);
```

Maybe we should remove these `await` for telemetry?